standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1505,14 +1507,9 @@ public void visit(LeafNode node) throws MetaException {
           params.add(catName.toLowerCase());
         }
         tableValue += " then " + tableValue0 + " else null end)";
-
-        if (valType == FilterType.Date) {
-        	tableValue = dbType.toDate(tableValue);
-        } else if (valType == FilterType.Timestamp) {
-          tableValue = dbType.toTimestamp(tableValue);
-        }
       }
-      if (!node.isReverseOrder) {
+
+      if (!node.isReverseOrder && nodeValue != null) {
```

Comment:
why do we need 2 checks, can we change to
````
if (nodeValue != null) { 
    params.add(nodeValue);
}
````

Reply:
it's used to avoid adding the same parameter twice to the statement, 
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java#L1482
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java#L1519

Reply:
I mean, instead of
````
if (node.isReverseOrder && nodeValue != null) {
        params.add(nodeValue);
}
....
if (!node.isReverseOrder && nodeValue != null) {
     params.put("catName", catName);
 }
````
can't we refactor to
````
if (nodeValue != null) { 
    params.add(nodeValue); 
}
````

Reply:
No.   As in the result filter, there could be some "?" in the `tableValue`,
```
String filter = node.isReverseOrder
              ? nodeValue0 + " " + node.operator.getSqlOp() + " " + tableValue
              : tableValue + " " + node.operator.getSqlOp() + " " + nodeValue0;
```
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java#L1502-L1508

We need to place the `nodeValue` at the right position in `params` according to the `isReverseOrder`

Reply:
👍